java-checks/src/main/java/org/sonar/java/checks/serialization/SerializableFieldInSerializableClassCheck.java

```diff
@@ -94,7 +95,7 @@ private static boolean isUnserializableCollection(Type type) {
   private void checkCollectionAssignments(List<IdentifierTree> usages) {
     for (IdentifierTree usage : usages) {
       Tree parentTree = usage.parent();
-      if (parentTree.is(Tree.Kind.ASSIGNMENT)) {
+      if (Objects.requireNonNull(parentTree).is(Tree.Kind.ASSIGNMENT)) {
```

Comment:
This is not a required change. Is it IntelliJ who suggests you to do it?

On this, We (as SonarSource) are not aligned with their definition of nullness. `Tree.Parent()` is annotated with `@Nullable`, which is a *weak* nullness annotation, contrarily to `@CheckForNull`. Here, it's the responsibility of the developer calling the method to know when it will be null or not, and it usually depends on very precise conditions, and it can not happen here. Our NPE detection rule is not raising any issue, and I believe we are right.

Could you revert the change?